src/briefcase/integrations/subprocess.py

```diff
@@ -26,6 +31,53 @@ def json_parser(json_output):
         raise ParseError(f"Failed to parse output as JSON: {e}")
 
 
+def is_process_dead(pid: int):
+    """
+    Returns True if a PID is not assigned to a process.
+
+    Checking if a PID exists is only a semi-safe proxy to determine
+    if a process is dead since PIDs can be re-used. Therefore, this
+    function should only be used via constant monitoring of a PID
+    to identify when the process goes from existing to not existing.
+
+    :param pid: integer value to be checked if assigned as a PID.
+    :return: True if PID doesn't exist; defaults to False if PID=None.
+    """
+    if pid is None:
+        return False
```

Comment:
Is this a good default? If the process dies really quickly, it won't get found on the PID search; so the PID will be None; this would seem to suggest that it will report as "process is not dead".

Reply:
Yeah...my thinking was that if the caller doesn't know the process (as made evident by the absent PID), then it may be better to assume the process exists. This approach also leans towards maintaining previous behavior by log streaming regardless of the App's status as running or not.

That said, this default was a late addition to this PR; I think I'd rather push the interpretation of `pid=None` back up to the caller; they will have better context on the appropriate behavior.